Source/WebCore/platform/graphics/cocoa/UnrealizedCoreTextFont.cpp

```diff
@@ -306,15 +306,29 @@ void UnrealizedCoreTextFont::modifyFromContext(const FontDescription& fontDescri
 
     m_size = getSize();
 
-    if (fontDescription.textRenderingMode() == TextRenderingMode::OptimizeLegibility)
-        m_opticalSizingType = OpticalSizingType::Everything;
-    else if (fontDescription.opticalSizing() == FontOpticalSizing::Disabled)
-        m_opticalSizingType = OpticalSizingType::None;
-    else
-        m_opticalSizingType = OpticalSizingType::JustVariation;
-
     m_variationSettings = fontDescription.variationSettings();
 
+    if (fontDescription.textRenderingMode() == TextRenderingMode::OptimizeLegibility) {
+        std::optional<float> opticalSizingValue;
+#if USE(CORE_TEXT_OPTICAL_SIZING_WORKAROUND)
+        // We can delete this when rdar://problem/104370451 is fixed.
+        // And then we can change OpticalSizingType back to an enum instead of a variant.
+        auto predicate = [opsz = fontFeatureTag("opsz")](const FontVariationSettings::Setting& setting) {
+            return setting.tag() == opsz;
+        };
+        auto iterator = std::find_if(m_variationSettings.begin(), m_variationSettings.end(), WTFMove(predicate));
```

Comment:
Free form `std::begin` and `std::end` are more general than class-based one (but it's not part of the WebKit guidelines though so I guess it's not important)

Reply:
Done.